insights/client/crypto.py

```diff
@@ -0,0 +1,122 @@
+import os.path
+import logging
+import shutil
+import tempfile
+import shlex
+import subprocess
+
+from .constants import InsightsConstants as constants
+
+logger = logging.getLogger(__file__)
+
+
+def _call_gpg(home, command):
+    """
+    Run /usr/bin/gpg.
+
+    Parameters:
+        home (str): Path to GPG's home directory.
+        command (str): Actual arguments passed to the GPG binary.
+
+    Returns (str, str, int):
+        Standard output, standard error and the return code.
+    """
+    command = (
+        "/usr/bin/gpg --homedir {home} {command}"
+    ).format(home=home, command=command)
+    process = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
It's always better to use a list to begin with rather than trying to split a string when calling subprocess.  That way things like spaces in filenames are treated correctly.
```suggestion
    command = (
        "/usr/bin/gpg", "--homedir", home, command
    )
    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```